src/lib/pickles_types/poly_types.ml

```diff
@@ -1,23 +1,36 @@
+(** The following module types define the different kinds of terms at the type
+    level by their arity. These types serves are manipulated and passed around as
+    parameters of the various functors defined in {!Hlist}.
+
+    Each term is represented by a module containing a single n-ary type.
+*)
+
+(** The kind of values (no type parameters). *)
 module type T0 = sig
   type t
 end
 
+(** Unary type functions (1 type parameter). *)
```

Comment:
why "type functions" ? That's a module type to me

Reply:
They **are** modules type but I argue that their purpose is that of type-level functions. For example,
```
type 'a t = ('a * int)
```
can be seen as a function `t` which takes a type `'a` (its parameter) and transforms it into a new type `('a * int)`, just like a normal function would do with values. And that's pretty much *how* these types are later used, to *map* a type to another type.

I could add these short explanation into the documentation to make things clearer.

Reply:
ah I see! Interesting :)

Reply:
It would be good to mention this analogy explicitly in the exposition at the top of the file